src/codegen/compvars.cpp

```diff
@@ -870,6 +921,13 @@ class IntType : public ConcreteCompilerType {
         return rtn;
     }
 
+    CompilerVariable* getiter(IREmitter& emitter, const OpInfo& info, VAR* var) override {
+        ConcreteCompilerVariable* converted = var->makeConverted(emitter, BOXED_INT);
+        CompilerVariable* rtn = converted->getiter(emitter, info);
+        converted->decvref(emitter);
+        return rtn;
```

Comment:
Yes, this is really causing a lot of boilerplate code :-(
If I convert to unknown wouldn't I then loose the type info? --> getType()->getattrType always would return unknown?